File: frontend/src/features/settings/sections/billing/googleReviewUrl.ts
/** Google g.page review link helpers (registration + billing settings). */

const GOOGLE_REVIEW_URL_RE = /^https:\/\/g\.page\/r\/[A-Za-z0-9_\-]+\/review\/?$/i;

export function normalizeGoogleReviewUrl(value: string | null | undefined): string {
  const normalized = (value || "").trim();
  if (!normalized) return "";
  return normalized.replace(/\/+$/, "");
}

export function isValidGoogleReviewUrl(value: string | null | undefined): boolean {
  const normalized = normalizeGoogleReviewUrl(value);
  if (!normalized) return false;
  return GOOGLE_REVIEW_URL_RE.test(`${normalized}/`);
}

File: backend/apps/settings/domain/google_review_url.py
# backend/apps/settings/domain/google_review_url.py
# Feladat: Google vélemény (g.page) link normalizálása és validálása.
# Sárközi Mihály - 2026.07.20

from __future__ import annotations

import re

_GOOGLE_REVIEW_URL_RE = re.compile(
    r"^https://g\.page/r/[A-Za-z0-9_\-]+/review/?$",
    re.IGNORECASE,
)


def normalize_google_review_url(value: str | None) -> str:
    normalized = (value or "").strip()
    if not normalized:
        return ""
    return normalized.rstrip("/")


def is_valid_google_review_url(value: str | None) -> bool:
    normalized = normalize_google_review_url(value)
    if not normalized:
        return False
    return bool(_GOOGLE_REVIEW_URL_RE.match(f"{normalized}/"))


__all__ = ["is_valid_google_review_url", "normalize_google_review_url"]
